extensions/ql-vscode/src/variant-analysis/variant-analysis-monitor.ts

```diff
@@ -60,20 +60,28 @@ export class VariantAnalysisMonitor extends DisposableObject {
 
     this.monitoringVariantAnalyses.add(variantAnalysis.id);
     try {
-      await this._monitorVariantAnalysis(variantAnalysis);
+      await this._monitorVariantAnalysis(
+        variantAnalysis.id,
+        variantAnalysis.controllerRepo.id,
+        variantAnalysis.executionStartTime,
+        variantAnalysis.query.name,
+        variantAnalysis.language,
+      );
     } finally {
       this.monitoringVariantAnalyses.delete(variantAnalysis.id);
     }
   }
 
   private async _monitorVariantAnalysis(
-    variantAnalysis: VariantAnalysis,
+    variantAnalysisId: number,
+    controllerRepoId: number,
+    executionStartTime: number,
+    queryName: string,
+    language: QueryLanguage,
   ): Promise<void> {
```

Comment:
Maybe a silly question: why has this changed to take all these bits of information instead of just `variantAnalysis`?

Reply:
I wanted to avoid passing a variant analysis object (and calling it `variantAnalysis`) as I think it'd make the code confusing further down (when we get the "actual" variant analysis from the manager).

So it was an intentional change for readability rather than a need. This is subjective ofc, so we if we don't think it makes things more readable then we should pass a `variantAnalysis` (or maybe call it something else?).

I think variant analyses generally have some information that doesn't change (the contoller repo id, the query info, etc.) so we could think about introducing a new. type to cover that. I think it could be used in other places too (e.g. the mapping functions). But anyway, that's a different rabbit hole 😂

Reply:
Ah okay, that makes sense! I've definitely been getting confused about all the `variantAnalysis` vars floating around (and when they're being updated) 😵 
So I'm happy with this just accepting the "unchanging" properties!